BuildDBSwiftBindingTest/main.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  main.swift
+//  BuildDBSwiftBindingTest
+//
+//  Copyright © 2019 Apple Inc. All rights reserved.
+//
+
+import llbuild
+import Foundation
+
+let path = "/Users/benjaminherzog/Library/Developer/Xcode/DerivedData/TEMPApp-bairfqxgkpxvqbglrdmnhebkyovk/Build/Intermediates.noindex/XCBuildData/build.db"
+
+class Delegate: BuildDBDelegate {
```

Comment:
The delegate abstraction in BuildDB is a very low-level thing we use to enable a performance optimization. I don't think it actually makes sense to expose it out of the public BuildDB API.

What the public API should do is just automatically "intern" keys and do the translation under the covers so that the API is neater. We can still expose the KeyID so clients can use a more compact representation, but otherwise shouldn't need to expose the raw API.

@dmbryson can you sanity check my reasoning here ^^^^

Reply:
Yes, the delegate is intended to just provide performance minded mapping between the keys and an internal mapping.  In the current use, the engine keeps the mapping (and is the delegate).   For these bindings, I think it would be appropriate for the internal C++ side of the binding to create its own mapping/ids. The database level KeyIDs should never be returned from the DB object.

Reply:
By database level KeyID you mean the key that is part of the `key_names` table in the sqlite db itself right? So my changes in BuildDB.h should be rather removed, right

Reply:
Correct